Add explicit METHOD to SOLVE if target is a DERIVATIVE block

JCGoran opened the pull request:
Fixes #3601.

Commit:
Add explicit METHOD to SOLVE if solving DERIVATIVE

Commit:
Update tests

JCGoran committed:
Update test/nmodl/transpiler/unit/visitor/implicit_method.cpp

cattabiani reviewed (dismissed):
I think the code can be improved (in efficiency. It works already)

Commit:
Implement feedback

- rename to SolveWithoutMethodVisitor
- use `count` instead of a linear search
- remove extra parentheses

Commit:
Merge remote-tracking branch 'origin/jelic/fix_implicit_solve_method' into jelic/fix_implicit_solve_method

nrnhines committed:
Merge branch 'master' into jelic/fix_implicit_solve_method
